core/src/main/java/org/frankframework/pipes/IfPipe.java

```diff
@@ -0,0 +1,277 @@
+import org.frankframework.core.PipeLineSession;
+import org.frankframework.core.PipeRunException;
+import org.frankframework.core.PipeRunResult;
+import org.frankframework.doc.EnterpriseIntegrationPattern;
+import org.frankframework.doc.Forward;
+import org.frankframework.parameters.ParameterList;
+import org.frankframework.stream.Message;
+import org.frankframework.util.MessageUtils;
+import org.frankframework.util.TransformerPool;
+import org.frankframework.util.XmlEncodingUtils;
+import org.frankframework.util.XmlUtils;
+
+@Forward(name = "*", description = "when {@literal thenForwardName} or {@literal elseForwardName} are used")
+@Forward(name = "then", description = "the configured condition is met")
+@Forward(name = "else", description = "the configured condition is not met")
+@EnterpriseIntegrationPattern(EnterpriseIntegrationPattern.Type.ROUTER)
+public class IfPipe extends AbstractPipe {
+
+	private String elseForwardName = "else";
+	private String expressionValue = null;
+	private String namespaceDefs = null;
+	private String thenForwardName = "then";
+	private boolean namespaceAware = XmlUtils.isNamespaceAwareByDefault();
+	private DefaultMediaType defaultMediaType = DefaultMediaType.XML;
+
+	public enum DefaultMediaType {
+		XML(MediaType.APPLICATION_XML),
+		JSON(MediaType.APPLICATION_JSON);
+
+		final MediaType mediaType;
+
+		DefaultMediaType(MediaType mediaType) {
+			this.mediaType = mediaType;
+		}
+	}
+
+	private TransformerPool transformerPool;
+	private String jsonPathExpression = null;
+	private String regex = null;
+	private String xpathExpression = null;
+	private int xsltVersion = XmlUtils.DEFAULT_XSLT_VERSION;
+
+	@Override
+	public void configure() throws ConfigurationException {
+		super.configure();
+
+		if (StringUtils.isNotEmpty(xpathExpression)) {
+			try {
+				transformerPool = TransformerPool.getInstance(makeStylesheet(xpathExpression, expressionValue), xsltVersion, this);
+			} catch (TransformerConfigurationException e) {
+				throw new ConfigurationException("could not create transformer from xpathExpression [" + xpathExpression + "], target expressionValue [" + expressionValue + "]", e);
+			}
+		}
+	}
+
+	@Override
+	public PipeRunResult doPipe(Message message, PipeLineSession session) throws PipeRunException {
+		MimeType mimeType = getMimeType(message);
+
+		// Determine the media type and if the correct *pathExpression was provided
+		if (jsonPathExpression != null && !MediaType.APPLICATION_JSON.equals(mimeType)
+				|| xpathExpression != null && !MediaType.APPLICATION_XML.equals(mimeType)) {
+			throw new PipeRunException(this, "Incorrect pathExpression provided for given mediaType + " + mimeType);
+		}
+
+		String forward = determineForward(message, session);
+
+		log.debug("determined forward [{}]", forward);
+
+		PipeForward pipeForward = findForward(forward);
+		if (pipeForward == null) {
+			throw new PipeRunException(this, "cannot find forward or pipe named [" + forward + "]");
+		}
```

Comment:
Dit kan je beter in de configure stap afvangen.

Reply:
Dit komt uit de XmlIf, wil je dat ik die ook aanpas?